File: QuickStart/ChartView/FirstLook/LegendElement.cs
using System;
using Telerik.WinControls.UI;
using System.Windows.Forms;
using System.Drawing;
using Telerik.WinControls;
using Telerik.Charting;

namespace Telerik.Examples.WinControls.ChartView.FirstLook
{
    public class LegendElement : LightVisualElement
    {
        StackLayoutElement layout;

        protected override void InitializeFields()
        {
            base.InitializeFields();

            this.Alignment = ContentAlignment.MiddleCenter;
            this.DrawFill = true;
            this.BackColor = Color.White;
            this.GradientStyle = GradientStyles.Solid;
            this.Padding = new Padding(5, 5, 5, 0);
            this.Margin = new Padding(0);
            this.MinSize = new Size(100, 0);
            this.StretchVertically = false;
            this.StretchHorizontally = true;
            this.ShouldHandleMouseInput = false;
            this.NotifyParentOnMouseInput = true;
        }

        protected override void CreateChildElements()
        {
            base.CreateChildElements();

            layout = new StackLayoutElement();
            layout.Orientation = Orientation.Vertical;
            layout.FitInAvailableSize = true;
            layout.StretchVertically = true;
            layout.StretchHorizontally = true;
            this.Children.Add(layout);
        }

        public void BuildLegend(RadChartElement element)
        {
            Telerik.WinControls.UI.ChartView view = element.View;
            BuildLegendCore(view);
        }

        void BuildLegendCore(Telerik.WinControls.UI.ChartView view)
        {
            layout.Children.Clear();
            if (view.AreaType == ChartAreaType.Cartesian)
            {
                foreach (Telerik.WinControls.UI.ChartSeries series in view.Series)
                {
                    Color color;
                    if(series is LineSeries)
                    {
                        color = series.BorderColor;
                    }
                    else
                    {
                        color = series.Children[0].BackColor;
                    }
                    layout.Children.Add(new LegendItemElement(series.Name, color));
                }
            }
            if (view.AreaType == ChartAreaType.Pie && view.Series.Count > 0)
            {
                for (int i = 0; i < view.Series[0].DataPoints.Count; i++)
                {
                    Color color = view.Series[0].Children[i].BackColor;
                    PieDataPoint point = view.Series[0].DataPoints[i] as PieDataPoint;
                    layout.Children.Add(new LegendItemElement(point.Name, color));
                }
            }
        }

        protected override SizeF MeasureOverride(SizeF availableSize)
        {
            SizeF size = base.MeasureOverride(availableSize);
            return size;
        }
    }
}

File: QuickStart/ChartView/FirstLook/LegendElement.vb
Imports System
Imports Telerik.WinControls.UI
Imports System.Windows.Forms
Imports System.Drawing
Imports Telerik.WinControls
Imports Telerik.Charting

Namespace Telerik.Examples.WinControls.ChartView.FirstLook
	Public Class LegendElement
		Inherits LightVisualElement
		Private layout As StackLayoutElement

		Protected Overrides Sub InitializeFields()
			MyBase.InitializeFields()

			Me.Alignment = ContentAlignment.MiddleCenter
			Me.DrawFill = True
			Me.BackColor = Color.White
			Me.GradientStyle = GradientStyles.Solid
			Me.Padding = New Padding(5, 5, 5, 0)
			Me.Margin = New Padding(0)
			Me.MinSize = New Size(100, 0)
			Me.StretchVertically = False
			Me.StretchHorizontally = True
			Me.ShouldHandleMouseInput = False
			Me.NotifyParentOnMouseInput = True
		End Sub

		Protected Overrides Sub CreateChildElements()
			MyBase.CreateChildElements()

			layout = New StackLayoutElement()
			layout.Orientation = Orientation.Vertical
			layout.FitInAvailableSize = True
			layout.StretchVertically = True
			layout.StretchHorizontally = True
			Me.Children.Add(layout)
		End Sub

		Public Sub BuildLegend(ByVal element As RadChartElement)
			Dim view As Telerik.WinControls.UI.ChartView = element.View
			BuildLegendCore(view)
		End Sub

		Private Sub BuildLegendCore(ByVal view As Telerik.WinControls.UI.ChartView)
			layout.Children.Clear()
			If view.AreaType = ChartAreaType.Cartesian Then
				For Each series As Telerik.WinControls.UI.ChartSeries In view.Series
					Dim color As Color
					If TypeOf series Is LineSeries Then
						color = series.BorderColor
					Else
						color = series.Children(0).BackColor
					End If
					layout.Children.Add(New LegendItemElement(series.Name, color))
				Next series
			End If
			If view.AreaType = ChartAreaType.Pie AndAlso view.Series.Count > 0 Then
				For i As Integer = 0 To view.Series(0).DataPoints.Count - 1
					Dim color As Color = view.Series(0).Children(i).BackColor
					Dim point As PieDataPoint = TryCast(view.Series(0).DataPoints(i), PieDataPoint)
					layout.Children.Add(New LegendItemElement(point.Name, color))
				Next i
			End If
		End Sub

		Protected Overrides Function MeasureOverride(ByVal availableSize As SizeF) As SizeF
			Dim size As SizeF = MyBase.MeasureOverride(availableSize)
			Return size
		End Function
	End Class
End Namespace
